2.0/gradle/vividus-test-docker.gradle

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+buildscript {
+    repositories {
+        gradlePluginPortal()
+    }
+    dependencies {
+        classpath(group: 'com.bmuschko', name: 'gradle-docker-plugin', version: '9.3.4')
+    }
+}
+
+apply plugin: com.bmuschko.gradle.docker.DockerRemoteApiPlugin
+
+tasks.register('copyResources', Copy) {
+    into layout.buildDirectory.dir('docker/resources')
+    from sourceSets.main.getResources().srcDirs
+}
+
+tasks.register('createTestsDockerfile', com.bmuschko.gradle.docker.tasks.image.Dockerfile) {
+    def dependencies = project.configurations.implementation.allDependencies;
+    def matcher = { dependency, targetName ->
+        dependency.group.equals('org.vividus') && dependency.name.equals(targetName)
+    }
+
+    def vividusLib = dependencies.find { matcher.call(it, 'vividus') }
+
+    def vividusVersion = vividusLib.version
+    if (vividusVersion == null) {
+        def vividusBomLib = dependencies.find { matcher.call(it, 'vividus-bom') }
+        vividusVersion = vividusBomLib.version
+    }
+
+    if (vividusVersion == null) {
+        throw new StopExecutionException('Unable to evaluate VIVIDUS version')
+    }
+
+    dependsOn copyResources
+    from("vividus/vividus:${vividusVersion}")
+    copyFile('resources', '/vividus/resources/')
+}
+
+tasks.register('buildTestsImage', com.bmuschko.gradle.docker.tasks.image.DockerBuildImage) {
+    dependsOn createTestsDockerfile
+
+    def repositoryPath = System.getenv('DOCKER_REPOSITORY_PATH')
+    if (repositoryPath == null) {
+        throw new StopExecutionException('The DOCKER_REPOSITORY_PATH is not set')
+    }
+
+    images.add("${repositoryPath}/${project.name}:${project.version}")
+}
+
+tasks.register('pushTestsImage', com.bmuschko.gradle.docker.tasks.image.DockerPushImage) {
+    dependsOn buildTestsImage
+    images = buildTestsImage.images
+
+    def registryUsername = System.getenv('DOCKER_REGISTRY_USERNAME')
+    def registryPassword = System.getenv('DOCKER_REGISTRY_PASSWORD')
+
+    if (registryUsername != null && registryPassword == null
+            || registryUsername == null && registryPassword != null) {
+        throw new StopExecutionException('The DDOCKER_REGISTRY_USERNAME and DOCKER_REGISTRY_PASSWORD env '
+            + 'variables should be either both set or unset')
```

Comment:
need to make more clear, like `Environment variable DOCKER_REGISTRY_USERNAME is set, but 
DOCKER_REGISTRY_PASSWORD is not` and vice versa